### PR TITLE
datetime to astropy Time: sunpy.timeseries

### PR DESCRIPTION
**Pull request by vn-ki**

### CODE REVIEW

File: sunpy/timeseries/sources/goes.py [hunk 1]
```diff
@@ -173,8 +174,10 @@ def _parse_hdus(cls, hdulist):
         else:
             raise ValueError("Don't know how to parse this file")
 
-        times = [start_time + datetime.timedelta(seconds=int(np.floor(s)),
-                                                    microseconds=int((s - np.floor(s)) * 1e6)) for s in seconds_from_start]
+        times = [start_time + TimeDelta(int(np.floor(s))*u.second +
+                                        int((s - np.floor(s)) * 1e6))*u.microsecond
+                                        for s in seconds_from_start]
+                                        # Why not just pass s???
```
Comment (nabobalis) [inline sunpy/timeseries/sources/goes.py]: Is the result the same if you just pass in `seconds_from_start` instead of having the `for` statement?

**Comment by nabobalis** [comment]
CircleCi conflict?

**Comment by nabobalis** [comment]
@pep8speaks quiet

File: sunpy/instr/fermi.py [hunk 2]
```diff
@@ -467,8 +467,8 @@ def met_to_utc(timeinsec):
     # The reference time for this is 2001-Jan-01 00:00.
     met_ref_time = parse_time('2001-01-01 00:00')
     offset_from_utc = (
-        met_ref_time - parse_time('1979-01-01 00:00')).total_seconds()
-    time_in_utc = parse_time(timeinsec + offset_from_utc)
+        met_ref_time - parse_time('1979-01-01 00:00')).sec
+    time_in_utc = parse_time(timeinsec + offset_from_utc, format='utime')
```
Comment (Cadair) [inline sunpy/instr/fermi.py]: can this be:

```python
return met_ref_time + timeinsec * u.s
```

File: sunpy/instr/fermi.py [hunk 3]
```diff
@@ -489,9 +489,9 @@ def utc_to_met(time_ut):
 
     """
     met_ref_time = parse_time('2001-01-01 00:00')
-    ut_seconds = (time_ut - parse_time('1979-01-01')).total_seconds()
+    ut_seconds = (time_ut - parse_time('1979-01-01')).sec
```
Comment (Cadair) [inline sunpy/instr/fermi.py]: same here, do we need to be doing all this `ut_seconds` stuff?

File: sunpy/instr/fermi.py [hunk 4]
```diff
@@ -449,7 +449,7 @@ def separation_angle(radec1, radec2):
 
 def met_to_utc(timeinsec):
     """
-    Converts Fermi Mission Elapsed Time (MET) in seconds to a datetime object.
+    Converts Fermi Mission Elapsed Time (MET) in seconds to a Time object.
```
Comment (Cadair) [inline sunpy/instr/fermi.py]: ``` `~astropy.time.Time` ```

File: sunpy/instr/fermi.py [hunk 5]
```diff
@@ -459,39 +459,32 @@ def met_to_utc(timeinsec):
 
     Returns
     -------
-    `datetime.datetime`
-        The input Fermi Mission Elapsed Time converted to a datetime object.
+    `astropy.time.Time`
+        The input Fermi Mission Elapsed Time converted to a Time object.
```
Comment (Cadair) [inline sunpy/instr/fermi.py]: same as above

File: sunpy/timeseries/metadata.py [hunk 6]
```diff
@@ -177,13 +177,13 @@ def append(self, timerange, columns, metadata, **kwargs):
...
-        for datetime and/or column name.
+        for `~astropy.time.Time` and/or column name.
         Will return all metadata entry indices if no filters are given.
 
         Parameters
         ----------
-        time : `str` or `~datetime.datetime` optional
-            The string (parsed using the `~sunpy.time.parse_time`) or datetime
+        time : `str` or `astropy.time.Time` optional
+            The string (parsed using the `~sunpy.time.parse_time`) or `Time`
```
Comment (Cadair) [inline sunpy/timeseries/metadata.py]: ``` `~astropy.time.Time` ```

File: sunpy/timeseries/metadata.py [hunk 7]
```diff
@@ -208,7 +208,7 @@ def find_indices(self, time=None, colname=None, **kwargs):
         # Find all results with suitable timerange.
         results_indices = []
         for i, meta in enumerate(self.metadata):
-            if dt in meta[0] or not(dt):
+            if not(dt) or dt in meta[0]:
```
Comment (Cadair) [inline sunpy/timeseries/metadata.py]: I think you could drop the brackets here. i.e. `not dt or dt in meta[0]`

File: sunpy/timeseries/sources/goes.py [hunk 8]
```diff
@@ -173,8 +174,8 @@ def _parse_hdus(cls, hdulist):
         else:
             raise ValueError("Don't know how to parse this file")
 
-        times = [start_time + datetime.timedelta(seconds=int(np.floor(s)),
-                                                    microseconds=int((s - np.floor(s)) * 1e6)) for s in seconds_from_start]
+        times = [start_time + TimeDelta(s*u.second) for s in seconds_from_start]
```
Comment (Cadair) [inline sunpy/timeseries/sources/goes.py]: This needs to be converted to `datetime` before it is passed to `DataFrame` on line 188 below.

Comment (Cadair) [inline sunpy/timeseries/sources/goes.py]: Also I don't think you need to loop here, off the top of my head I think this should work:
```python
times = start_time + TimeDelta(seconds_from_start*u.s)
```

File: sunpy/instr/fermi.py (hunk 5 above)
Comment (vn-ki) [reply sunpy/instr/fermi.py]: But it's already mentioned once, right? Should we do it again?

File: sunpy/timeseries/metadata.py (hunk 7 above)
Comment (vn-ki) [reply sunpy/timeseries/metadata.py]: How about `(not dt)`, just to be extra clear?

File: sunpy/timeseries/metadata.py [hunk 9]
```diff
@@ -221,13 +221,13 @@ def find_indices(self, time=None, colname=None, **kwargs):
...
         """
-        Find all metadata matching the given filters for datetime and/or column name.
+        Find all metadata matching the given filters for `~astropy.time.Time` and/or column name.
         Will return all metadata entries if no filters are given.
 
         Parameters
         ----------
-        time : `str` or `~datetime.datetime` optional
-            The string (parsed using the `~sunpy.time.parse_time`) or datetime
+        time : `str` or `astropy.time.Time` optional
```
Comment (nabobalis) [inline sunpy/timeseries/metadata.py]: Do we want to `~` all `astropy.time.Time` as a rule?

File: sunpy/timeseries/sources/goes.py [hunk 10]
```diff
@@ -173,8 +174,8 @@ def _parse_hdus(cls, hdulist):
...
+                                        # Why not just pass s???
```
Comment (nabobalis) [inline sunpy/timeseries/sources/goes.py]: Also remove this comment?

File: sunpy/timeseries/metadata.py (hunk 9 above)
Comment (vn-ki) [reply sunpy/timeseries/metadata.py]: I thought seeing just `Time` would be confusing for new users.

Comment (Cadair) [reply sunpy/timeseries/metadata.py]: I think it's context dependant. I am in favour in full pathing it in this bit of the docstring.

File: sunpy/timeseries/sources/goes.py (hunk 8 above)
Comment (vn-ki) [reply sunpy/timeseries/sources/goes.py]: That is so beautiful!! :heart_eyes_cat:

File: sunpy/instr/fermi.py [hunk 11]
```diff
@@ -459,39 +459,32 @@ def met_to_utc(timeinsec):
...
     ----------
-    time_ut : 'datetime.datetime'
-        A datetime object in UT
+    time_ut : `astropy.time.Time`
+        A `~astropy.time.Time` object in UT
 
     Returns
     -------
-    'float'
+    `astropy.units.second`
```
Comment (Cadair) [inline sunpy/instr/fermi.py]: This should be `astropy.units.Quantity`

**Pull request merged**